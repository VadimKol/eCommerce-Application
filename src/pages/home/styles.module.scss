.main {
  padding: 20px;
}

.container {
  display: flex;
  flex-direction: column;
  gap: 10px;
  width: 100%;
  min-width: 300px;
  max-width: 100%;
  margin: auto;
  padding: 28px 40px 30px;
  text-align: center;
  color: var(--primary-color);
  background:
    linear-gradient(var(--background-color) 0 0) padding-box,
    linear-gradient(45deg, var(--primary-color), var(--secondary-color)) border-box;
  border: 5px solid transparent;
  border-radius: 10px;
}

.title {
  margin-block: 0;
}

.linksDesc {
  margin: 0;
}

.linksList {
  display: flex;
  flex-direction: column;
  gap: 10px;
  margin: 0;
  padding: 0;
  list-style: none;
}
