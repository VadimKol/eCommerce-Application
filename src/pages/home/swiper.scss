.swiper {
  min-height: 100px;
}

.pageSwiper {
  max-width: 100%;
  max-height: inherit;
}

.modalSwiper {
  max-width: 100%;
  height: 100%;
  background: linear-gradient(var(--background-color) 0 0) padding-box, linear-gradient(45deg, var(--primary-color), var(--secondary-color)) border-box;
  border: 5px solid transparent;
  border-radius: 10px;
}

.swiper-wrapper {
  max-height: inherit;

  .modalSwiper & {
    max-height: 55svh;
  }
}

.swiper-slide {
  display: flex;
  justify-content: center;
  align-items: center;
  height: auto;
  min-height: 200px;
  background-size: cover;
}

.swiper-button-prev,
.swiper-button-next {
  text-shadow: 0 0 4px 4px var(--background-color);
  transition: color 0.3s ease-in-out;

  @media (hover: hover) {
    &:hover,
    &:focus-visible {
      color: var(--tertiary-color);
    }
  }
}

.swiper-pagination-bullet {
  --swiper-pagination-bullet-inactive-opacity: 0.5;
  --swiper-pagination-bullet-inactive-color: var(--secondary-color);

  width: 10px;
  height: 10px;
  box-shadow: 0 0 4px 4px var(--background-color);
  transition: background-color 0.3s ease-in-out;

  @media (hover: hover) {
    &:hover,
    &:focus-visible {
      background: var(--tertiary-color);
    }
  }
}

.swiper-pagination-bullet-active:hover {
  cursor: default;
}

.hide,
.link,
.modalSwiper a {
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
}

.avatar {
  background-image: url("../../assets/images/avatar_banner.png");
}

.war {
  background-image: url("../../assets/images/war_banner.png");
}


@media (width <= 700px) {
  .swiper-slide {
    background-position: 60% 50%;
  }

  .modalSwiper {
    height: 300px;
  }
}
