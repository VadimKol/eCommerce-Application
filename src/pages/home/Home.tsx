import 'swiper/css';
import 'swiper/css/navigation';
import 'swiper/css/pagination';
import './swiper.scss';

import classNames from 'classnames';
import { Navigation, Pagination } from 'swiper/modules';
import { Swiper, SwiperSlide } from 'swiper/react';

import styles from './styles.module.scss';

export function Home(): JSX.Element {
  const images = [
    '../../assets/images/promo_banner.png',
    '../../assets/images/avatar_banner.png',
    '../../assets/images/war_banner.png',
  ];
  return (
    <main className={classNames('main', styles.main)}>
      <div className={styles.container}>
        <h1 className={styles.title}>Home page</h1>
        <Swiper
          initialSlide={0}
          spaceBetween={10}
          slidesPerView={1}
          navigation
          lazyPreloadPrevNext={1}
          loop
          pagination={{
            clickable: true,
          }}
          modules={[Navigation, Pagination]}
          className="modalSwiper"
        >
          {images &&
            images.map((image, index) => (
              <SwiperSlide key={image} className={styles.swiperSlide}>
                <img src={image} alt={`banner number ${index + 1}`} className={styles.modalImage} loading="lazy" />
                <div className="swiper-lazy-preloader" />
              </SwiperSlide>
            ))}
        </Swiper>
        <div className={styles.animateBlock}>
          <div className={classNames(styles.animateLine, styles.animateLineFirst)}>
            Apply Promo Code &quot;GEEK-SHOP&quot; and get 10% discount
          </div>
          <div className={classNames(styles.animateLine, styles.animateLineSecond)}>
            Apply Promo Code &quot;GEEK-SHOP&quot; and get 10% discount
          </div>
        </div>
      </div>
    </main>
  );
}
