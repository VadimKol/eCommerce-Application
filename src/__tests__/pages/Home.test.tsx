import { render } from '@testing-library/react';

import { Home } from '@/pages/home/Home';

jest.mock('swiper/css', () => '');
jest.mock('swiper/css/navigation', () => '');
jest.mock('swiper/css/pagination', () => '');
jest.mock('swiper/modules', () => ({
  Navigation: jest.fn(),
  Pagination: jest.fn(),
  Autoplay: jest.fn(),
}));
jest.mock('swiper/react', () => ({
  Swiper: jest.fn(),
  SwiperSlide: jest.fn(),
}));

it('renders correctly', () => {
  const { container } = render(<Home />);

  expect(container).toMatchSnapshot();
});
