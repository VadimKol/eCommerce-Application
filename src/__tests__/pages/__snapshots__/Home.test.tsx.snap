// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <main
    class="main main"
  >
    <div
      class="animateBlock green"
    >
      <div
        class="animateLine"
      >
        Click and get the Promo Code "COSPLAY" with a 10% discount on the category
         
      </div>
    </div>
    <div
      class="animateBlock tertiary"
    >
      <div
        class="animateLine"
      >
        Click and get the Promo Code "GEEK-SHOP" and get 10% discount for all
         
      </div>
    </div>
    <div
      class="container"
    >
      <h1
        class="title"
      >
        Home page
      </h1>
      <h2
        class="h2Title"
      >
        Our Products 
      </h2>
      <div
        class="catalogBlock"
      >
        <div
          class="catalogItem"
        >
          <a
            class="link"
            href="/catalog/cosplay/weapon"
          >
            <img
              alt="Weapons"
              class="catalogImage"
              src="stub-test"
            />
            <h3
              class="catalogTitle"
            >
              Weapons
            </h3>
          </a>
        </div>
        <div
          class="catalogItem"
        >
          <a
            class="link"
            href="/catalog/cosplay/wigs"
          >
            <img
              alt="Wigs"
              class="catalogImage"
              src="stub-test"
            />
            <h3
              class="catalogTitle"
            >
              Wigs
            </h3>
          </a>
        </div>
        <div
          class="catalogItem"
        >
          <a
            class="link"
            href="/catalog/decor/figures"
          >
            <img
              alt="Figures"
              class="catalogImage"
              src="stub-test"
            />
            <h3
              class="catalogTitle"
            >
              Figures
            </h3>
          </a>
        </div>
        <div
          class="catalogItem"
        >
          <a
            class="link"
            href="/catalog/decor/mugs"
          >
            <img
              alt="Mugs"
              class="catalogImage"
              src="stub-test"
            />
            <h3
              class="catalogTitle"
            >
              Mugs
            </h3>
          </a>
        </div>
        <div
          class="catalogItem"
        >
          <a
            class="link"
            href="/catalog/decor/pads"
          >
            <img
              alt="Pads"
              class="catalogImage"
              src="stub-test"
            />
            <h3
              class="catalogTitle"
            >
              Pads
            </h3>
          </a>
        </div>
        <div
          class="catalogItem"
        >
          <a
            class="link"
            href="/catalog/stationery/notepads"
          >
            <img
              alt="Notepads"
              class="catalogImage"
              src="stub-test"
            />
            <h3
              class="catalogTitle"
            >
              Notepads
            </h3>
          </a>
        </div>
        <div
          class="catalogItem catalogItemMore"
        >
          <a
            class="link"
            href="/catalog"
          >
            <h3
              class="catalogTitle"
            >
              Show more at Catalog
            </h3>
          </a>
        </div>
      </div>
    </div>
  </main>
</div>
`;
